doc/api/inspector.md

```diff
@@ -379,6 +379,61 @@ session.post('HeapProfiler.takeHeapSnapshot', null, (err, r) => {
 });
 ```
 
+### Class: `inspector.RawSession`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* Extends: {EventEmitter}
+
+The `inspector.RawSession` APIs are similar to `inspector.Session`. But the
+data is dispatching to V8 directly and the data returned by V8 is also returned
+to the user directly.
```

Comment:
This isn't really explaining what the _specific_ differences between the two APIs would be, though. "directly" is a relative word; there can be more or less abstraction in an API, but the question that the docs should answer (and which would also help determine whether this is a reasonable feature) is what the actual level of difference is. For example: What can you do with this API that you can't with the regular `Session` API? What does the `Session` API do that this one doesn't?.

At a glance, it seems like the two APIs let callers achieve the same goals.

Reply:
Thanks, i will update it later.

Reply:
I see that this has been updated, but it’s still not really clear what you can do with this API that you can’t with the regular `Session`.

As far as I can tell, this PR would be equivalent to:

```js
class RawSession extends EventEmitter {
  constructor(normalSession) {
    super();
    this._session = normalSession;
    this._session.on('inspectorNotification', (parsed) => {
      this.emit('message', JSON.stringify(parsed));
    });
  }

  connect() {
    this._session.connect();
  }

  connectToMainThread() {
    this._session.connectToMainThread();
  }

  post(message) {
    const { method, params } = JSON.parse(message);
    this._session.post(method, params, (err, result) => {
      if (err) {
        if (err.code === 'ERR_INSPECTOR_COMMAND') {
          const error = err.message.match(
            /^Inspector error (?<code>[^:]+): (?<message>.+)$/).groups;
          this.emit('message', JSON.stringify({ error }));
        } else {
          this.emit('message', JSON.stringify(err));
        }
      } else {
        this.emit('message', JSON.stringify({ result }));
      }
    })
  }

  disconnect() {
    this._session.disconnect();
  }
}
```

Reply:
Yes, but a little different. I think the `post` is as follows.
```js
const idCallbackMap = {};
post(message) {
  const { id, method, params } = JSON.parse(message);
  const cb = (err, result) => {
    // send to clien
    socket.send(JSON.stringify({
      id,
      err,
      result
    }));
    delete idCallbackMap[id];
  }
  idCallbackMap[id] = cb;
  this._session.post(method, params, cb)
}
```

What i think is like this.

```js
const { RawSession } = require('inspector');
const { WebSocketServer } = require('ws');

const server = new WebSocketServer({port: 8080});

const session = new RawSession();

// Or call connectToMainThread in worker
session.connect();

server.on('connection', (connection) => {
    connection.on('message', (data) => {
        session.post(data.toString());
    });
});

session.on('message', (result) => {
    connection.send(result);
});

```